src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseDetails.java

```diff
@@ -330,6 +330,21 @@ private void setAfterSubmitCallbackResponseEntity(final AfterSubmitCallbackRespo
         return metadata;
     }
 
+    @JsonIgnore
+    public Map<String, Object> getFormattedMetadata() {
```

Comment:
do we need completely new method for this, can't we change this in getMetaData method

Reply:
When getFormattedMetaData is called before getMetaData then it will populate metadata map already.
it may break other functionalities.

if you are confident that functionality is not going to break, then in both methods, only below line is different, 
            metadata.put(CASE_REFERENCE.getReference(), getReference() != null ? getReference() : getId());
            
            Which can be handled by boolean parameter, Think about this